promql/parse.go

```diff
@@ -933,14 +941,18 @@ func (p *parser) offset() time.Duration {
 //		[<metric_identifier>] <label_matchers>
 //
 func (p *parser) VectorSelector(name string) *VectorSelector {
-	var matchers []*labels.Matcher
+	ret := &VectorSelector{
+		Name: name,
+	}
 	// Parse label matching if any.
 	if t := p.peek(); t.typ == LEFT_BRACE {
-		matchers = p.labelMatchers(EQL, NEQ, EQL_REGEX, NEQ_REGEX)
+		p.generatedParserResult = ret
+
+		p.parseGenerated(START_LABELS, []ItemType{RIGHT_BRACE, EOF})
```

Comment:
Do we have a unittest covering a } inside a label value?

Reply:
You mean something like `foo{bar="}"}` ?

Reply:
Yes

Reply:
AFAIK we don't. But if something like that would cause problems it would be a lexer bug.

The parser only sees the token stream:

    IDENTIFIER "foo"
    LEFT_BRACE "{"
    IDENTIFIER "bar"
    EQL "="
    STRING "\"}\""
    RIGHT_BRACE "}"

Reply:
It doesn't hurt to have a unittest to be safe.

Reply:
I've added a test case.